ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/Genesis/Tests/LongRangeAttack.hs

```diff
@@ -23,17 +23,21 @@ import           Test.Util.Orphans.IOLike ()
 import           Test.Util.TestBlock (TestBlock, unTestHash)
 
 tests :: TestTree
-tests = testProperty "long range attack" prop_longRangeAttack
+tests =
+  testGroup "long range attack" [
+    testProperty "one adversary" (prop_longRangeAttack 1 [10]),
+    testProperty "three adversaries" (prop_longRangeAttack 1 [2, 5, 10])
+  ]
 
 genChainsAndSchedule :: Word -> ScheduleType -> QC.Gen (GenesisTest, PointSchedule)
 genChainsAndSchedule numAdversaries scheduleType =
   unsafeMapSuchThatJust do
     gt <- genChains numAdversaries
     pure $ ((gt,) <$> genSchedule scheduleType (gtBlockTree gt))
 
-prop_longRangeAttack :: QC.Gen QC.Property
-prop_longRangeAttack = do
-  (genesisTest, schedule) <- genChainsAndSchedule 1 FastAdversary
+prop_longRangeAttack :: Int -> [Int] -> QC.Gen QC.Property
+prop_longRangeAttack honestFreq advFreqs = do
+  (genesisTest, schedule) <- genChainsAndSchedule (fromIntegral (length advFreqs)) (Frequencies freqs)
```

Comment:
Haskell question: why is `fromIntegral` necessary here?

Reply:
`genChainsAndSchedule` takes a `Word`

Reply:
Ah dayum.